hack/make/build-rpm

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+set -e
+
+DEST=$1
+
+# subshell so that we can export PATH without breaking other things
+(
+	source "$(dirname "$BASH_SOURCE")/.integration-daemon-start"
+
+	# TODO consider using frozen images for the dockercore/builder-rpm tags
+		
+	rpmVersion="${VERSION//-/_}"
+	# if we have a "-dev" suffix or have change in Git, let's make this package version more complex so it works better
+	if [[ "$VERSION" == *-dev ]] || [ -n "$(git status --porcelain)" ]; then
+		gitUnix="$(git log -1 --pretty='%at')"
+		gitDate="$(date --date "@$gitUnix" +'%Y%m%d.%H%M%S')"
+		gitCommit="$(git log -1 --pretty='%h')"
+		gitVersion="git${gitDate}.0.${gitCommit}"
+		# gitVersion is now something like 'git20150128.112847.0.17e840a'
+		rpmVersion="${rpmVersion}_${gitVersion}"
```

Comment:
The Fedora packaging guides cover that. I would likely put it in a '.'
after the release.

cc @lsm5
On May 1, 2015 5:12 PM, "Tianon Gravi" <notifications@github.com> wrote:

> In hack/make/build-rpm
> <https://github.com/docker/docker/pull/12917#discussion_r29534108>:
>
> > +
> > +# subshell so that we can export PATH without breaking other things
> > +(
> > +	source "$(dirname "$BASH_SOURCE")/.integration-daemon-start"
> > +
> > +	# TODO consider using frozen images for the dockercore/builder-rpm tags
> > +		
> > +	rpmVersion="${VERSION//-/_}"
> > +	# if we have a "-dev" suffix or have change in Git, let's make this package version more complex so it works better
> > +	if [[ "$VERSION" == *-dev ]] || [ -n "$(git status --porcelain)" ]; then
> > +		gitUnix="$(git log -1 --pretty='%at')"
> > +		gitDate="$(date --date "@$gitUnix" +'%Y%m%d.%H%M%S')"
> > +		gitCommit="$(git log -1 --pretty='%h')"
> > +		gitVersion="git${gitDate}.0.${gitCommit}"
> > +		# gitVersion is now something like 'git20150128.112847.0.17e840a'
> > +		rpmVersion="${rpmVersion}_${gitVersion}"
>
> @vbatts <https://github.com/vbatts> will
> 1.7.0_dev_git20150128.112847.0.17e840a be preferred over just a bare 1.7.0,
> or will the bare version take precedence? how are rc, beta, alpha, and/or
> nightly releases usually handled in RPM land to make sure that 1.7.0
> replaces 1.7.0_rc1?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/12917/files#r29534108>.
>